user_interaction_txt_text.py

```diff
@@ -0,0 +1,94 @@
+import os
+from enum import Enum, auto
+from typing import Optional
+
+from user_file_manager import UserFileManager
+
+
+class MessageScope(Enum):
+    GLOBAL = auto()
+    ACTIVE = auto()
+    PASSIVE = auto()
+
+
+class UserInteraction():
+    def __init__(self, context: 'Context'):
+        self.context = context
+        self.user_names = self.context.user_names
+
+        self.file_manager = UserFileManager(
+            os.path.join(os.path.dirname(__file__), "History"))
+        self.game = self.context._game
+        self._prepared_messages = {
+            MessageScope.GLOBAL: [],
+            MessageScope.ACTIVE: [],
+            MessageScope.PASSIVE: []
+        }
+
+    def save_global(self, text: str) -> None:
+        self._prepared_messages[MessageScope.GLOBAL].append(text)
+
+    def save_active(self, text: str) -> None:
+        self._prepared_messages[MessageScope.ACTIVE].append(text)
+
+    def save_passive(self, text: str) -> None:
+        self._prepared_messages[MessageScope.PASSIVE].append(text)
+
+    def show_global_instant(self, text: str) -> None:
+        print(f"{MessageScope.GLOBAL.name}: {text}")
+        self.file_manager.update_history_text(self.user_names[0], text)
+        self.file_manager.update_history_text(self.user_names[1], text)
+
+    def show_active_instant(self, text: str) -> None:
+        print(self.user_names[0])
+        print(f"{MessageScope.ACTIVE.name}: {text}")
+        self.file_manager.update_history_text(
+            self.game.active_player.user_name, text)
+
+    def show_passive_instant(self, text: str) -> None:
+        print(self.user_names[1])
+        print(f"{MessageScope.PASSIVE.name}: {text}")
+        self.file_manager.update_history_text(
+            self.game.passive_player.user_name, text)
+
+    def show_all(self) -> None:
+        for scope, text_list in self._prepared_messages.items():
+            print(f"{scope.name}:")
+            for text in text_list:
+                print(f"\t{text}")
+                if scope == MessageScope.GLOBAL:
+                    self.file_manager.update_history_text(
+                        self.game.active_player.user_name, text)
+                    self.file_manager.update_history_text(
+                        self.game.passive_player.user_name, text)
+
+                elif scope == MessageScope.ACTIVE:
+                    self.file_manager.update_history_text(
+                        self.game.active_player.user_name, text)
+
+                else:
+                    self.file_manager.update_history_text(
+                        self.game.passive_player.user_name, text)
+
+        self._clear_messages()
+
+    def _clear_messages(self) -> None:
+        self._prepared_messages = {
+            MessageScope.GLOBAL: [],
+            MessageScope.ACTIVE: [],
+            MessageScope.PASSIVE: []
+        }
+
+    def read_number(self, text: str = "") -> Optional[int]:
```

Comment:
Code duplication with original UserInteraction.

Could be avoided by either creating of base classes for both `UserInteraction` classes, or by delegation some functionaliry from original `UserInteraction` class

Reply:
Done